util/tele/span_logger.go

```diff
@@ -88,14 +88,18 @@ func (s *spanLogSink) Error(err error, msg string, keysAndValues ...interface{})
 	)
 }
 
-func (s *spanLogSink) WithValues(keysAndValues ...interface{}) logr.LogSink {
-	s.vals = append(s.vals, keysAndValues...)
```

Comment:
The issue with this is that `s` is being mutated through the pointer receiver, so this slice will grow forever and can't be garbage collected. The method receiver here is changed to a value, which fixes the memory issue.

Reply:
This is how the default `klog` implementation is set up: https://github.com/kubernetes/klog/blob/75663bb798999a49e3e4c0f2375ed5cca8164194/klogr.go#L94